### PR TITLE
Sync parameter names and function types with spec

### PR DESCRIPTION
**Pull request by pcnfernando**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.regexp/src/main/ballerina/regexp.bal [hunk 1]
```diff
@@ -38,21 +38,21 @@ type GroupsAsSpanArrayType SpanAsTupleType[];
 type GroupsArrayType GroupsAsSpanArrayType[];
 
 # Returns the span of the first match that starts at or after startIndex.
-public function find(RegExp reExp, string str, int startIndex = 0) returns Span? {
-    SpanAsTupleType? resultArr = findImpl(reExp, str, startIndex);
+public isolated function find(RegExp 're, string str, int startIndex = 0) returns Span? {
```
Comment (pcnfernando) [inline langlib/lang.regexp/src/main/ballerina/regexp.bal:41]: Depends on https://github.com/ballerina-platform/ballerina-spec/issues/1130#issuecomment-1275568944

File: langlib/lang.regexp/src/main/ballerina/regexp.bal [hunk 2]
```diff
@@ -174,30 +175,30 @@ public function replace(RegExp reExp, string str, Replacement replacement, int s
...
-function replaceFromString(RegExp reExp, string str, string replacementStr, int startIndex) returns string = @java:Method {
+isolated function replaceFromString(RegExp reExp, string str, string replacementStr, int startIndex) returns string = @java:Method {
     'class: "org.ballerinalang.langlib.regexp.Replace",
     name: "replaceFromString"
 } external;
 
 # Replaces all occurrences of a regular expression.
-public function replaceAll(RegExp reExp, string str, Replacement replacement, int startIndex = 0) returns string {
-    Groups? findResult = findGroups(reExp, str, startIndex);
+public isolated function replaceAll(RegExp 're, string str, Replacement replacement, int startIndex = 0) returns string {
```
Comment (pcnfernando) [inline langlib/lang.regexp/src/main/ballerina/regexp.bal:187]: Diff for not using `@isolatedParam` in the parameter depends on https://github.com/ballerina-platform/ballerina-spec/issues/1130#issuecomment-1275589846

**Comment by gimantha** [comment]
Shall we push this changes to master also? @pcnfernando

**Comment by pcnfernando** [comment]
> Shall we push this changes to master also? @pcnfernando

Created https://github.com/ballerina-platform/ballerina-lang/pull/38196

**Pull request merged**